web-components/src/components/advance-list/AdvanceList.ts

```diff
@@ -0,0 +1,268 @@
+import { Key } from "@/constants";
+import { customElementWithCheck } from "@/mixins/CustomElementCheck";
+import { FocusMixin } from "@/mixins/FocusMixin";
+import reset from "@/wc_scss/reset.scss";
+import { LitElement, PropertyValues, html, internalProperty, property, query, queryAll } from "lit-element";
+import { scroll } from "lit-virtualizer";
+import styles from "./scss/module.scss";
+
+export namespace AdvanceList {
+  export const prefixId = "item-";
+  @customElementWithCheck("md-advance-list")
+  export class ELEMENT extends FocusMixin(LitElement) {
+    @property({ type: Array }) items: any[] = [];
+    @property({ type: Boolean }) isLoading = false;
+    @property({ type: String }) selectedItemId = "";
+    @property({ type: String }) value = "";
+    @property({ type: String }) ariaRoleList = "";
+    @property({ type: String }) ariaLabelList = "";
+    @property({ type: Boolean }) isError = false;
+    @queryAll("div.default-wrapper") lists?: HTMLDivElement[];
+    @query(".virtual-scroll") listContainer?: HTMLDivElement;
+    @internalProperty() page = 1;
+    @property({ type: Number }) totalRecords = 0;
+    @internalProperty() scrollIndex = -1;
+    @internalProperty() activeId: string = "";
+    @internalProperty() isUserNavigated = false; // this flag is used to control scroll to index this will became true only when user navigated using keyboard
+
+    connectedCallback(): void {
+      super.connectedCallback();
+      this.addEventListener("click", this.handleClick);
+    }
+
+    static get styles() {
+      return [reset, styles];
+    }
+
+    disconnectedCallback() {
+      super.disconnectedCallback();
+      // Clean up event listeners when the component is removed
+      this.removeEventListener("click", this.handleClick);
+      this.listContainer?.removeEventListener("keydown", this.handleKeyDown.bind(this));
+    }
+
+    protected firstUpdated(_changedProperties: PropertyValues): void {
+      // Add keydown event listener to the list container
+      this.listContainer?.addEventListener("keydown", this.handleKeyDown.bind(this));
```

Comment:
Why we are using bind in the function. We should not use it. You can make the function as arrow function which will do the same job as bind.